File: 720-longest-word-in-dictionary/longest-word-in-dictionary.java
class Solution {
    public String longestWord(String[] words) {
        Arrays.sort(words);
        Set<String> builtWords = new HashSet<>();
        builtWords.add("");
        String longest = "";

        for (String word : words) {
            String prefix = word.substring(0, word.length() - 1);
            if (builtWords.contains(prefix)) {
                builtWords.add(word);
                if (word.length() > longest.length()) {
                    longest = word;
                }
            }
        }

        return longest;
    }
}

File: 720-longest-word-in-dictionary/longest-word-in-dictionary.py
class Solution:

  def longestWord(self, words: list[str]) -> str:
    words.sort()
    built_words = {""}
    longest = ""

    for word in words:
      if word[:-1] in built_words:
        built_words.add(word)
        if len(word) > len(longest):
          longest = word

    return longest